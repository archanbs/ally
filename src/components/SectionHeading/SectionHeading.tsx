import React from "react";

export type SectionHeadingProps = {
  label: string;
  selected?: boolean;
  id?: string;
  onClick?: () => void;
};

export const SectionHeading = ({ label, selected, id, onClick }: SectionHeadingProps) => {
  const customStyle = {
    borderRight: "none",
  };

  return (
    <div
      className="py-4 border-b border-r border-gray-300"
      style={selected ? customStyle : undefined}
      id={id}
      onClick={(ev) => {
        ev.preventDefault();
        onClick();
      }}
    >
      {label}
    </div>
  );
};
