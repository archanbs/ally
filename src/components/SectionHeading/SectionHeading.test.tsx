import React from "react";
import { SectionHeading } from "./SectionHeading";
import renderer from "react-test-renderer";
import { fireEvent, render, screen } from "@testing-library/react";

test("Section heading is rendered correctly", () => {
  const compTree = renderer.create(<SectionHeading label="heading" />).toJSON();

  expect(compTree).toMatchSnapshot();
});

test("Section heading is rendered correctly with selected", () => {
  const compTree = renderer.create(<SectionHeading label="heading" selected />).toJSON();

  expect(compTree).toMatchSnapshot();
});

test("Section heading click triggers the callback function", async () => {
  const obj = {
    clickCallback: function () {
      console.log("click callback called");
    },
  };
  jest.spyOn(obj, "clickCallback");

  render(<SectionHeading label="heading" selected onClick={obj.clickCallback} />);
  const testElement = await screen.findByText("heading");
  expect(testElement).toBeInTheDocument();

  fireEvent.click(testElement);
  expect(obj.clickCallback).toHaveBeenCalled();
});
