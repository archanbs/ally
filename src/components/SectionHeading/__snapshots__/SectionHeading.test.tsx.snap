// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section heading is rendered correctly 1`] = `
<div
  className="py-4 border-b border-r border-gray-300"
  onClick={[Function]}
>
  heading
</div>
`;

exports[`Section heading is rendered correctly with selected 1`] = `
<div
  className="py-4 border-b border-r border-gray-300"
  onClick={[Function]}
  style={
    Object {
      "borderRight": "none",
    }
  }
>
  heading
</div>
`;
